### PR TITLE
chore: create endpoint to retrieve wallet balance

### PR DESCRIPTION
**Pull request by Leona-cloud**

### CODE REVIEW

File: src/modules/api/wallet/services/index.ts [hunk 1]
```diff
@@ -1715,31 +1716,69 @@ export class WalletService {
         });
     }
 
-    async getTotalWalletBalance() {
-        const totalBalance = await this.prisma.wallet.aggregate({
+    private async aggregateTotalWalletBalance() {
```
Comment (calculusky) [inline src/modules/api/wallet/services/index.ts:1719]: This should return an defined interface for commission balance and main balance. Also specify the return type in the method declaration

File: src/modules/api/wallet/services/index.ts [hunk 2]
```diff
@@ -1715,31 +1716,69 @@ export class WalletService {
...
-                commissionBalance: totalBalance._sum.commissionBalance || 0,
+                mainWalletBalance: mainWalletBalance || 0,
+                mainCommissionBalance: mainCommissionBalance || 0,
+                walletOpeningBalance: openingBalance.main || 0,
+                commissionOpeningBalance: openingBalance.commission || 0,
             },
         });
     }
+
+    @Cron("59 23 * * *", {
```
Comment (calculusky) [inline src/modules/api/wallet/services/index.ts:1755]: Create a scheduler module and implement your cron logic

File: src/modules/api/wallet/services/index.ts [hunk 3]
```diff
@@ -1716,20 +1715,25 @@ export class WalletService {
...
+    async aggregateTotalWalletBalance(): Promise<WalletBalance> {
+        const result = await this.prisma.wallet.aggregate({
             _sum: {
                 commissionBalance: true,
                 mainBalance: true,
             },
         });
+
+        return {
+            commissionBalance: result._sum.commissionBalance,
```
Comment (calculusky) [inline src/modules/api/wallet/services/index.ts:1727]: You didn't add check if sum is null

File: src/modules/api/wallet/services/index.ts [hunk 4]
```diff
@@ -1716,20 +1715,25 @@ export class WalletService {
...
+            mainBalance: result._sum.mainBalance,
+        };
     }
 
     async getTotalWalletBalance() {
         const mainWalletBalance = (await this.aggregateTotalWalletBalance())
```
Comment (calculusky) [inline src/modules/api/wallet/services/index.ts:1733]: Please improve readability here

File: src/modules/scheduler/services/index.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        private walletService: WalletService
+    ) {}
+    @Cron("59 23 * * *", {
+        timeZone: "Africa/Lagos",
+    })
+    async aggregateWalletOpeningBalance() {
+        try {
+            const mainWalletBalance = (
+                await this.walletService.aggregateTotalWalletBalance()
+            ).mainBalance;
```
Comment (calculusky) [inline src/modules/scheduler/services/index.ts:20]: Improve readability here

File: src/modules/scheduler/services/index.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                update: {
+                    main: mainWalletBalance,
+                    commission: mainCommissionBalance,
+                },
+                create: {
+                    main: mainWalletBalance,
+                    commission: mainCommissionBalance,
+                },
+            });
+            console.log("Wallet opening balance updated successfully");
```
Comment (calculusky) [inline src/modules/scheduler/services/index.ts:37]: Remove all console logs

File: src/modules/scheduler/services/index.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+export class SchedulerService {
+    constructor(
+        private prisma: PrismaService,
+        @Inject(forwardRef(() => WalletService))
+        private walletService: WalletService
+    ) {}
+    @Cron("59 23 * * *", {
+        timeZone: "Africa/Lagos",
+    })
+    async aggregateWalletOpeningBalance() {
```
Comment (calculusky) [inline src/modules/scheduler/services/index.ts:16]: Did you test this function to run automatically at set interval?

Comment (Leona-cloud) [reply src/modules/scheduler/services/index.ts:16]: Yes set the time to 5 and 45seconds respectively. the function ran at the specified time.

**Pull request merged**